### ISSUE TITLE
`OverflowError` with Contrast Adjustment panel for constant images

### ISSUE THREAD

**Issue by gprevot** [PlotPyStack/PlotPy#25]
When an image containing a constant value is selected for displaying level histograms, the x axis limits of the LevelsHistogram plot are set to a very small range, i.e. xmin and xmax are nearly the same. This cause self.range (XRangeSelection) to be drawn very far in canvas frame coordinates, which cause a failure in XRangeSelection.draw(), for

 painter.drawLine(center_x, top, center_x, bottom)

as center_x may be an integer below -2147483648 or above 2147483647, which may be not the case
it can be corrected by changing this line to: 

painter.drawLine(QPointF(center_x, top), QPointF(center_x, bottom))

**Comment by PierreRaybaut** [comment]
Thanks for the feedback.
I can't reproduce the issue on Windows at least (e.g. when modifying the `plotpy/tests/features/test_contrast.py` example and replacing the image data by a constant array:

```python
def test_contrast3():
    """Contrast test 3

    Test if level histogram handles correctly the case where the image values are
    constant
    """
    with qt_app_context(exec_loop=True):
        item = make.image(np.ones((100, 100), dtype=np.uint16) * 1.234)
        _win = __create_dialog_with_contrast(item)

if __name__ == "__main__":
    test_contrast3()
```

However, your suggestion of modification is coherent with the code above in `plotpy\items\shape\range.py`, so I'm merging your code change, hoping that it will fix this on your side.

**Comment by gprevot** [comment]
Thanks!

By the way, there is a small mistake for histogram panel when used  
with masked images. If self.min_select_tool
or self.max_select_tool are applied on a masked point, then it  
converts the masked value to nan, which causes an error:

C:\Users\33699\anaconda3\lib\site-packages\plotpy\items\image\base.py:510:  
UserWarning: Warning: converting a masked element to nan.
   fmin, fmax = float(self.min), float(self.max)  # avoid overflows
C:\Users\33699\anaconda3\lib\site-packages\qwt\interval.py:54:  
UserWarning: Warning: converting a masked element to nan.
   self.__minValue = float(minValue)  # avoid overflows with NumPy scalars
C:\Users\33699\anaconda3\lib\site-packages\plotpy\items\shape\range.py:102:  
UserWarning: Warning: converting a masked element to nan.
   rct2.setLeft(xMap.transform(self._min))
Traceback (most recent call last):

   File ~\anaconda3\lib\site-packages\qwt\plot_canvas.py:636 in paintEvent
     self.drawCanvas(painter, False)

   File ~\anaconda3\lib\site-packages\qwt\plot_canvas.py:700 in drawCanvas
     self.plot().drawCanvas(painter)

   File ~\anaconda3\lib\site-packages\qwt\plot.py:1359 in drawCanvas
     self.drawItems(painter, self.__data.canvas.contentsRect(), maps)

   File ~\anaconda3\lib\site-packages\qwt\plot.py:1383 in drawItems
     item.draw(painter, maps[item.xAxis()], maps[item.yAxis()], canvasRect)

   File ~\anaconda3\lib\site-packages\plotpy\items\shape\range.py:113 in draw
     center_x = int(rct2.center().x())

ValueError: cannot convert float NaN to integer


best regards

Geoffroy


Pierre Raybaut ***@***.***> a écrit :

> Thanks for the feedback.
> I can't reproduce the issue on Windows at least (e.g. when modifying  
> the `plotpy/tests/features/test_contrast.py` example and replacing  
> the image data by a constant array:
>
> ```python
> def test_contrast3():
>     """Contrast test 3
>
>     Test if level histogram handles correctly the case where the  
> image values are
>     constant
>     """
>     with qt_app_context(exec_loop=True):
>         item = make.image(np.ones((100, 100), dtype=np.uint16) * 1.234)
>         _win = __create_dialog_with_contrast(item)
>
> if __name__ == "__main__":
>     test_contrast3()
> ```
>
> However, your suggestion of modification is coherent with the code  
> above in `plotpy\items\shape\range.py`, so I'm merging your code  
> change, hoping that it will fix this on your side.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/PlotPyStack/PlotPy/issues/25#issuecomment-2386025209
> You are receiving this because you authored the thread.
>
> Message ID: ***@***.***>



-- 
Geoffroy Prévot
Directeur Adjoint
Institut des NanoSciences de Paris
4, place Jussieu
75252 PARIS cedex 05
https://fr.linkedin.com/in/geoffroy-pr%C3%A9vot-493016233

**Comment by PierreRaybaut** [comment]
I can't reproduce the `ValueError` with masked images. Maybe it's fixed in latest release? (V2.6.3 released 5 minutes ago)

**Comment by gprevot** [comment]
Yes, I have updated to 2.6.3 and it is fixed!

Pierre Raybaut ***@***.***> a écrit :

> I can't reproduce the `ValueError` with masked images. Maybe it's  
> fixed in latest release? (V2.6.3 released 5 minutes ago)
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/PlotPyStack/PlotPy/issues/25#issuecomment-2386430484
> You are receiving this because you authored the thread.
>
> Message ID: ***@***.***>



-- 
Geoffroy Prévot
Directeur Adjoint
Institut des NanoSciences de Paris
4, place Jussieu
75252 PARIS cedex 05
https://fr.linkedin.com/in/geoffroy-pr%C3%A9vot-493016233

**Comment by PierreRaybaut** [comment]
> Yes, I have updated to 2.6.3 and it is fixed! Pierre Raybaut ***@***.***> a écrit :
> […](#)
> I can't reproduce the `ValueError` with masked images. Maybe it's fixed in latest release? (V2.6.3 released 5 minutes ago) -- Reply to this email directly or view it on GitHub: [#25 (comment)](https://github.com/PlotPyStack/PlotPy/issues/25#issuecomment-2386430484) You are receiving this because you authored the thread. Message ID: ***@***.***>
> -- Geoffroy Prévot Directeur Adjoint Institut des NanoSciences de Paris 4, place Jussieu 75252 PARIS cedex 05 https://fr.linkedin.com/in/geoffroy-pr%C3%A9vot-493016233

Great news, thanks for the feedback.
